cmd/podman/images/scan.go

```diff
@@ -0,0 +1,156 @@
+package images
+
+import (
+	"context"
+	"fmt"
+	"github.com/containers/podman/v2/cmd/podman/common"
+	"github.com/containers/podman/v2/cmd/podman/registry"
+	"github.com/containers/podman/v2/pkg/domain/entities"
+	"github.com/containers/podman/v2/pkg/specgen"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"os"
+	"strings"
+)
+
+var (
+	scanDescription = `Scan a container image for known vulnerabilities.  The image name or digest can be used.`
+	scanCommand     = &cobra.Command{
+		Use:     "scan [run-options] IMAGE [scanner-options]",
+		Short:   "Scan a container image for known vulnerabilities",
+		Long:    scanDescription,
+		RunE:    scan,
+		Example: `podman scan centos:latest`,
+	}
+
+	imageScanCommand     = &cobra.Command{
+		Use:     "scan [run-options] IMAGE [scanner-options]",
+		Short:   "Scan a container image for known vulnerabilities",
+		Long:    scanDescription,
+		RunE:    scan,
+		Example: `podman image scan centos:latest`,
+	}
+)
+
+var (
+	scanOptions = entities.ImageScanOptions{
+		ContainerRunOptions: entities.ContainerRunOptions{
+			OutputStream: os.Stdout,
+			ErrorStream:  os.Stderr,
+			Rm: true,
+		},
+	}
+
+	cliVals common.ContainerCLIOpts
+)
+
+func scanFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+
+	common.DefineCreateFlags(cmd, &cliVals)
+	common.DefineNetFlags(cmd)
+
+	// TODO: derive default from containers.conf configuration
+	scannerFlagName := "scanner"
+	flags.StringVar(&scanOptions.ScannerImage, scannerFlagName, "docker.io/library/grype:latest", "The vulnerability scanner container image to use")
+}
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: scanCommand,
+	})
+	scanFlags(scanCommand)
+
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: imageScanCommand,
+		Parent:  imageCmd,
+	})
+	scanFlags(imageScanCommand)
+
+}
+
+func scan(cmd *cobra.Command, args []string) error {
+	var err error
+
+	if len(args) < 1 {
+		return fmt.Errorf("requires exactly one image, given %+v", args)
+	}
+	var imageToScan = args[0]
+	var mountTarget = "/podman-scan-image-mount"
+	var scanMount = fmt.Sprintf("type=image,src=%s,target=%s", imageToScan, mountTarget)
```

Comment:
Should default to read/only.

Reply:
righto --the default for an `image` mount type is readonly (`readwrite=false`) but explicit is more clear anyway